ownCloud/Client/Sharing/PublicLinkTableViewController.swift

```diff
@@ -0,0 +1,273 @@
+				return share.type == .link
+			}
+			self.shares = sharesWithReshares
+			OnMainThread {
+				self.removeShareSections()
+				self.addShareSections()
+				self.handleEmptyShares()
+			}
+		}, keepRunning: true)
+		shareQuery?.refreshInterval = 2
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+		navigationController?.isToolbarHidden = true
+	}
+
+	override func viewDidAppear(_ animated: Bool) {
+		super.viewDidAppear(animated)
+		handleEmptyShares()
+	}
+
+	// MARK: - Sharing UI
+
+	func addSectionFor(type: OCShareType, with title: String) {
+		var shareRows: [StaticTableViewRow] = []
+
+		let user = shares.filter { (share) -> Bool in
+			return share.type == type
+		}
+
+		if user.count > 0 {
+			for share in user {
+				if canEdit(share: share) {
+					shareRows.append( StaticTableViewRow(rowWithAction: { [weak self] (_, _) in
+						guard let self = self else { return }
+						let editPublicLinkViewController = PublicLinkEditTableViewController(style: .grouped)
+						editPublicLinkViewController.share = share
+						editPublicLinkViewController.core = self.core
+						editPublicLinkViewController.item = self.item
+						self.navigationController?.pushViewController(editPublicLinkViewController, animated: true)
+					}, title: share.name!, subtitle: share.permissionDescription(), accessoryType: .disclosureIndicator) )
+				} else {
+					shareRows.append( StaticTableViewRow(rowWithAction: nil, title: share.name!, subtitle: share.permissionDescription(), accessoryType: .none) )
+				}
+			}
+		} else {
+			shareRows.append( StaticTableViewRow(buttonWithAction: { [weak self] (_, _) in
+				self?.addPublicLink()
+			}, title: "Create Public Link".localized, style: StaticTableViewRowButtonStyle.plain))
+		}
+
+		let sectionType = "share-section-\(String(type.rawValue))"
+		if let section = self.sectionForIdentifier(sectionType) {
+			self.removeSection(section)
+		}
+		let section : StaticTableViewSection = StaticTableViewSection(headerTitle: title, footerTitle: nil, identifier: sectionType, rows: shareRows)
+		self.addSection(section, animated: false)
+	}
+
+	func addShareSections() {
+		OnMainThread {
+			self.addSectionFor(type: .link, with: "Public Links".localized)
+		}
+	}
+
+	func removeShareSections() {
+		OnMainThread {
+			let types : [OCShareType] = [.link]
+			for type in types {
+				let identifier = "share-section-\(String(type.rawValue))"
+				if let section = self.sectionForIdentifier(identifier) {
+					self.removeSection(section)
+				}
+			}
+		}
+	}
+
+	func resetTable(showShares : Bool) {
+		removeShareSections()
+		if shares.count > 0 && showShares {
+			messageView?.message(show: false)
+		}
+		self.addShareSections()
+	}
+
+	func handleEmptyShares() {
+		if shares.count == 0 {
+			OnMainThread {
+				self.resetTable(showShares: false)
+			}
+		}
+	}
+
+	// MARK: - Private Link Section
+
+	func addPrivateLinkSection() {
+		let identifier = "private-link-section"
```

Comment:
Design-wise, I'd prefer this section to have a title "Private Link", the private link URL to be shown below it as regular row. And then "Copy" as a button on the right side in the same row. Feels a bit busy right now.